### PR TITLE
finder: tagged search tune (allow to set cost for tags to be less prefered)

### PR DESCRIPTION
**Pull request by msaf1980**
Allow to tune which tag will be used as filter in primary key

### CODE REVIEW

File: finder/tagged_test.go [hunk 1]
```diff
@@ -74,34 +75,139 @@ func TestParseSeriesByTag(t *testing.T) {
 	assert := assert.New(t)
 
 	ok := func(query string, expected []TaggedTerm) {
-		p, err := ParseSeriesByTag(query)
+		p, err := ParseSeriesByTag(query, nil)
 		assert.NoError(err)
-		assert.Equal(expected, p)
+		length := len(expected)
+		if length < len(p) {
```
Comment (Civil) [inline finder/tagged_test.go:81]: In what case len(expected) can be not equal to len(p)? As it seems that previously it was a valid condition and test should anyway end up with an error.

File: finder/tagged.go [hunk 2]
```diff
@@ -226,14 +262,54 @@ func ParseTaggedConditions(conditions []string) ([]TaggedTerm, error) {
...
+		sort.Sort(TaggedTermList(terms))
+	} else {
+		// compare with taggs costs
+		sort.Slice(terms, func(i, j int) bool {
+			eq, comparable := lessCosts(terms, i, j)
+			if comparable {
+				return eq
+			}
+
+			if terms[i].Op < terms[j].Op {
```
Comment (Civil) [inline finder/tagged.go:282]: I would like to see more comments about what this code should do. As I'm a bit lost trying to understand why comparing one TaggedTermOp with another is a good sorting condition (I'm actually wonder how good that clause is and how different Term would affect performance, as I'm not sure if Match vs Not Match matters, as well as Eq and Not Eq likely have same cost for us)

File: finder/tagged.go [hunk 3]
```diff
@@ -226,14 +262,54 @@ func ParseTaggedConditions(conditions []string) ([]TaggedTerm, error) {
...
+				return true
+			}
+			if terms[i].Op > terms[j].Op {
```
Comment (Civil) [inline finder/tagged.go:285]: Also, overall it seems that closes should be moved to the bottom of this function as it seems to be that line 305 shouldn't be reachable at all.

**Review by Civil — CHANGES_REQUESTED** [review]

File: finder/tagged_test.go (hunk 1 above)
Comment (msaf1980) [reply finder/tagged_test.go:81]: ParseSeriesByTag return a slice (p). If a bug, length can be non equal of length of expected results. `assert.Equal` produce not readable result for compare slices or map. But not required and can be removed.

Comment (Civil) [reply finder/tagged_test.go:81]: Yeah I understand that, but still that seems to be a good idea to produce an error here (maybe custom one if assert gives non-readable message).

File: finder/tagged.go (hunk 3 above)
Comment (msaf1980) [reply finder/tagged.go:285]: May be cleanup block and use `if terms[i].Op == terms[j].Op` for select code block ?

**Review by Civil — APPROVED** [review]

**Pull request merged**